internal/lox/scanner.go

```diff
@@ -28,7 +28,7 @@ type Scanner struct {
 	source  string
 	start   int
 	current int
-	line    int
```

Comment:
This is unnecessary complexity - would go back to the int version and init with 1 if easy enough to do

Reply:
@ryan-gang that's not quite what I meant btw - I meant that as long as line is initialized to 1 in the scanner (and it is never 0), automatically the other functions you have should just propagate the correct value to `EOF`. It's incorrect to set `EOF` to 1 explicitly. Can you take a look at this PR? https://github.com/codecrafters-io/interpreter-tester/pull/15